Mathlib/Data/List/OfFn.lean

```diff
@@ -181,6 +181,63 @@ theorem pairwise_ofFn {R : α → α → Prop} {n} {f : Fin n → α} :
     (Fin.rightInverse_cast (length_ofFn f)).surjective.forall, Fin.forall_iff, Fin.cast_mk,
     Fin.mk_lt_mk, forall_comm (α := (_ : Prop)) (β := ℕ)]
 
+lemma head_ofFn {n} (hn : n ≠ 0) (f : Fin n → α) :
+    (ofFn f).head (mt ofFn_eq_nil_iff.1 hn) = f ⟨0, Nat.pos_of_ne_zero hn⟩ := by
```

Comment:
You could write this as:
```lean 4
lemma head_ofFn {n} (h : ofFn f ≠ []) (f : Fin n → α) :
    (ofFn f).head h = f ⟨0, sorry⟩ := sorry
```
(replacing the sorry with a proof `0 < n` , of course). This would make it able to be used as a simp lemma.

Reply:
Likewise for `getLast_ofFn`.

Reply:
I've revised those two lemma statements as suggested.